be1-go/message/answer.go

```diff
@@ -86,6 +87,17 @@ func (e *Error) Error() string {
 	return e.Description
 }
 
+// Errorf returns an error with the new formatting
+func Errorf(format string, err error) error {
+	msgError := &Error{}
+
+	if xerrors.As(err, &msgError) {
+		msgError.Description = fmt.Sprintf(format, msgError.Description)
```

Comment:
This assumes `format` will only have one format specifier and `msg.Description` will always satisfy it. That's easy to overlook.

I'd suggest

```go
func NewError(description string, parent error) error {
    msgError := &Error{}
    
    if xerrors.As(parent, &msgError) {
        msgError.Description = fmt.Sprintf("%s: %s", description, msgError.Description)
        return msgError
    }

    return xerrors.Errorf("%s: %v", description, parent)
}
```

Sample usage:

```go
return message.NewError("failed to validate the data", err)
```

Reply:
Yes I agree! Thank you for the suggestion. I made the modifications.